tests/fuzz/fuzz-addr.c

```diff
@@ -15,7 +16,14 @@ void run(const uint8_t *data, size_t size)
 {
 	uint8_t *script_pubkey = tal_dup_arr(tmpctx, uint8_t, data, size, 0);
 
-	encode_scriptpubkey_to_addr(tmpctx, chainparams, script_pubkey);
+	char *addr = encode_scriptpubkey_to_addr(tmpctx, chainparams, script_pubkey);
+
+	uint8_t *decoded_script_pubkey;
+
+	if (addr) {
+		decode_scriptpubkey_from_addr(tmpctx, chainparams, addr, &decoded_script_pubkey);
+		assert(tal_arr_eq(script_pubkey, decoded_script_pubkey));
```

Comment:
We should check the returned bool as well:

```suggestion
		assert(decode_scriptpubkey_from_addr(tmpctx, chainparams, addr, &decoded_script_pubkey));
		assert(tal_arr_eq(script_pubkey, decoded_script_pubkey));
```